_posts/2017-01-30-docker-frontend-tools.md

```diff
@@ -0,0 +1,146 @@
+- A `docker-compose.yml` configuration and Docker commands for running theme commands such as `npm start` from within the container
+
+Along the way, I'll also provide:
+
+- A quick overview of why we use Docker for local development
+- Tips for building custom images and running common front-end applications inside containers.
+
+## Background
+
+We at Savas switched to using Docker for local development last year and we love it - so much so that we even proposed [a Drupalcon session on our approach and experience](https://events.drupal.org/node/17250). Using Docker makes it incredibly easy for developers to quickly spin up consistent local development environments that match production. In the past we used Vagrant and virtual machines for these purposes, but we've found Docker to be much faster when switching between multiple projects.
+
+Usually we build our Docker images from scratch to perfectly match production environments. However, for agile development and rapid prototyping, we sometimes make use of public Docker images. In these cases we've relied on [Wodby's Docker4Drupal project](https://github.com/wodby/docker4drupal), which is "a set of docker containers optimized for Drupal."
+
+We plan to post a more in-depth series of posts covering Docker for local development and CI environments in the near future, so be on the look out for that!
+
+We're also fans of using style guides and atomic design to drive front-end development. So when we heard of the [Pattern Lab Starter Drupal theme](https://github.com/phase2/pattern-lab-starter) which leverages [Pattern Lab](http://patternlab.io/) (a tool for building pattern-driven user interfaces using atomic design), we were very excited to try it out.
+
+## Building the Docker image
+
+To try out the Pattern Lab Starter theme we started with a fresh local Drupal 8 installation, and then quickly spun up our local Docker development environment [using Docker4Drupal](http://docs.docker4drupal.org/en/latest/#usage). We then copied the [Pattern Lab Starter code](https://github.com/phase2/pattern-lab-starter) to a new `custom/theme/patter_lab_starter` directory in our Drupal project.
+
+However, running the Phase 2 Pattern Lab Starter theme requires Node, npm, PHP, and Composer - the former for the node dependencies (Gulp, Bower, etc.) and the latter for running Pattern Lab.
+
+While we could install these dependencies locally, that defeats the purpose of using Docker. One of the great advantages of virtualization, be it Docker or a full VM, is that you don't have to rely on installing global dependencies on your local machine. One of the many benefits of this is that it ensures each team member is developing in the same environment.
+
+Unfortunately, while Docker4Drupal provides public images for many applications (such as Nginx, PHP, MariaDB, Mailhog, Redis, Apache Solr, and Varnish), it does not provide images for running the Pattern Lab Starter theme dependencies.
```

Comment:
Not a review comment, but questions for my learning: 
- Are "applications" and "dependencies" interchangeable in general? They basically both mean "software/programs" right?
- If a PHP container can be provided by Docker4Drupal, why can't Pattern Lab use that one in conjunction with a container `node-composer` instead of `node-php-composer`?